### PR TITLE
[ADF-2925] Required and invalid validators added

### PR DESCRIPTION
**Pull request by bbcodrin**
**Please check if the PR fulfills these requirements**

> - [X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [X] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [X] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
Is displayed the following incorrect message: "Invalid Format"
https://issues.alfresco.com/jira/browse/ADF-2925

**What is the new behaviour?**
The field is empty so should be displayed the following error message: "Required value"

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by eromano** [comment]
can you provide a unit test for this change?

**Comment by bbcodrin** [comment]
I added a test here: 
https://github.com/Alfresco/alfresco-ng2-components/pull/3279/files

**Comment by eromano** [comment]
The test has to be in the same PR can you move the test related to this PR here please?

File: lib/content-services/search/components/search-number-range/search-number-range.component.spec.ts [hunk 1]
```diff
@@ -124,4 +125,11 @@ describe('SearchNumberRangeComponent', () => {
         expect(context.queryFragments['range1']).toEqual('cm:content.size:<0 TO 100>');
     });
 
+    it('should throw "pattern" error if FROM value is formed by letters', () => {
+        component.from = new FormControl(123);
+
+        component.ngOnInit();
+
+        expect(component.from.hasError('pattern'));
```
Comment (mauriziovitale) [inline lib/content-services/search/components/search-number-range/search-number-range.component.spec.ts]: What are we trying to check? Expect what? to be true to be false?
I was expecting to have a unit test like 

component.from = new FormControl('abc');
expect(component.from.hasError('pattern').toBe(true));


component.from = new FormControl(123);
expect(component.from.hasError('pattern').toBe(false));

component.from = new FormControl();
expect(component.from.hasError('required').toBe(true));

component.from = new FormControl(123);
expect(component.from.hasError('required').toBe(false));

File: lib/content-services/search/components/search-number-range/search-number-range.component.spec.ts [hunk 2]
```diff
@@ -136,4 +136,23 @@ describe('SearchNumberRangeComponent', () => {
...
         expect(component.formValidator).toBeTruthy();
     });
+
+    it('should throw "pattern" error if FROM value is formed by letters', () => {
+        const validators = Validators.compose([
+           Validators.required,
+           Validators.pattern(/^-?(0|[1-9]\d*)?$/)
+        ]);
+
+        component.from = new FormControl('abc', validators);
```
Comment (mauriziovitale) [inline lib/content-services/search/components/search-number-range/search-number-range.component.spec.ts]: split in different unit tests

File: lib/content-services/search/components/search-number-range/search-number-range.component.spec.ts [hunk 3]
```diff
@@ -136,4 +136,23 @@ describe('SearchNumberRangeComponent', () => {
 
         expect(component.formValidator).toBeTruthy();
     });
+
+    it('should throw "pattern" error if FROM value is formed by letters', () => {
+        const validators = Validators.compose([
```
Comment (mauriziovitale) [inline lib/content-services/search/components/search-number-range/search-number-range.component.spec.ts]: we must use the Validator that is inside the ngOnInit

**Pull request merged**